agenda/data/src/main/java/com/gkp/agenda/data/workers/CreatedItemsWorker.kt

```diff
@@ -0,0 +1,54 @@
+package com.gkp.agenda.data.workers
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.gkp.agenda.data.getType
+import com.gkp.agenda.data.toReminderBody
+import com.gkp.agenda.data.toTaskBody
+import com.gkp.agenda.domain.datasource.LocalAgendaDataSource
+import com.gkp.agenda.domain.model.AgendaItem
+import com.gkp.agenda.domain.model.AgendaItemType
+import com.gkp.core.database.dao.CreatedAgendaItemsDao
+import com.gkp.core.network.TaskyRetrofitApi
+
+class CreatedItemsWorker(
+    private val taskyRetrofitApi: TaskyRetrofitApi,
+    private val localDataSource: LocalAgendaDataSource,
+    private val createdAgendaItemsDao: CreatedAgendaItemsDao,
+    context: Context,
+    params: WorkerParameters,
+) : CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+        val agendaItemId = inputData.getString(AGENDA_ITEM_ID) ?: return Result.failure()
+        val agendaType = inputData.getString(AGENDA_ITEM_TYPE) ?: return Result.failure()
+        return try {
+            val agendaItem = localDataSource.getAgendaItemById(agendaItemId)
+            performPushCreatedRequest(agendaItem)
+            createdAgendaItemsDao.deleteById(agendaItemId)
+
+            Result.success()
+        } catch (e: Exception) {
+            Result.failure()
+        }
```

Comment:
Here, you aren’t ever returning the `Result. retry()`, so this worker can either succeed or fail. You'll need to return that result to tell `WorkManager` that this work needs to be retried based on the retry policy you’ve set. Here is a useful extension function to return a worker result based on the network error received from your API request:

```
fun DataError.toWorkerResult(): ListenableWorker.Result {
    return when(this) {
        DataError.Local.DISK_FULL -> ListenableWorker.Result.failure()
        DataError.Network.REQUEST_TIMEOUT -> ListenableWorker.Result.retry()
        DataError.Network.UNAUTHORIZED -> ListenableWorker.Result.retry()
        DataError.Network.CONFLICT -> ListenableWorker.Result.retry()
        DataError.Network.TOO_MANY_REQUESTS -> ListenableWorker.Result.retry()
        DataError.Network.NO_INTERNET -> ListenableWorker.Result.retry()
        DataError.Network.PAYLOAD_TOO_LARGE -> ListenableWorker.Result.failure()
        DataError.Network.SERVER_ERROR -> ListenableWorker.Result.retry()
        DataError.Network.SERIALIZATION -> ListenableWorker.Result.failure()
        DataError.Network.UNKNOWN -> ListenableWorker.Result.failure()
    }
}
```

You’ll also need to check how often the work has been executed if you want to limit it to a certain amount of retries.

```
if(runAttemptCount >= 5) {
  return Result.failure()
}
```